### PR TITLE
Update TS CI

### PR DESCRIPTION
**Pull request by orouz**
- [x] build `tsocnfig.refs.json` once

see https://github.com/build-security/kibana/runs/4865756281?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/csp-temp-ci.yml [hunk 1]
```diff
@@ -25,8 +24,5 @@ jobs:
       - name: kbn bootstrap
         run: yarn kbn bootstrap
 
-      - name: check TS
-        run: node scripts/type_check
```
Comment (orouz) [inline .github/workflows/csp-temp-ci.yml:29]: 1. so turns out this happens in `bootstrap` already, no need to keep it here
2. we may update this again to only check for TS errors in our project to cut down on time length

**Comment by ari-aviran** [comment]
Looks like a TS compilation failure during the bootstrap does not fail the build, like [in here](https://github.com/build-security/kibana/runs/4865519383?check_suite_focus=true). Shouldn't we have a way to fail the build if TS compilation fails?

**Comment by orouz** [comment]
> Looks like a TS compilation failure during the bootstrap does not fail the build, like [in here](https://github.com/build-security/kibana/runs/4865519383?check_suite_focus=true). Shouldn't we have a way to fail the build if TS compilation fails?

yeah, seems  like it :) good catch! 

technically both steps build `tsconfig.refs.json` but the first catches errors and continues. 

i'll update the PR to either run only on our project or use the second second which errs correctly.

**Comment by orouz** [comment]
looks like the CI is failing correctly now, as `main` still has TS errors. 

the main change here is that _it should_ take less time to run, as we're running `--build` on a smaller `tsconfig.json`

**Review by ari-aviran — APPROVED** [review]

**Review by ari-aviran — APPROVED** [review]

**Review by ari-aviran — DISMISSED** [review]

**Review by DaveSys911 — APPROVED** [review]

**Pull request merged**